### ISSUE TITLE
Bug: `id-length` rule does not limit namespace import identifier

### ISSUE THREAD

**Issue by Tanujkanti4441** [eslint/eslint#18847]
### Environment

Node version: latest
npm version: latest
Local ESLint version: latest
Global ESLint version: none
Operating System: windows

### What parser are you using?

Default (Espree)

### What did you do?

</details>

```js
/*eslint id-length: ["error"]*/

import * as x from 'module';
```

### What did you expect to happen?

rule will report `x` as it's length is less than 2 (default minimum length).

### What actually happened?

rule didn't report the identifier.

### Link to Minimal Reproducible Example

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgaWQtbGVuZ3RoOiBbXCJlcnJvclwiXSovXG5cbmltcG9ydCAqIGFzIHggZnJvbSAnbW9kdWxlJzsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by JoshuaKGoldberg** [comment]
👍 from me on this, seems like directly a new syntax feature that should now be supported.

**Comment by Chaedie** [comment]
Hello @Tanujkanti4441 

I'm sorry I didn't catch up the process of PR (Issue -> Assign -> PR). 
So I already made a PR for this Issue. 

Could you please assign me?

**Comment by Tanujkanti4441** [comment]
> Hello @Tanujkanti4441 
> 
> I'm sorry I didn't catch up the process of PR (Issue -> Assign -> PR). 
> So I already made a PR for this Issue. 
> 
> Could you please assign me if you don't mind?

Sure! but first we have to wait for the acceptance of this issue.

**Comment by Chaedie** [comment]
@Tanujkanti4441 Thank you for your kindness. I will wait for the acceptance. 

Thanks.

**Comment by mdjermanovic** [comment]
I also think is a bug. This rule already checks default imports ([Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgaWQtbGVuZ3RoOiBbXCJlcnJvclwiXSovXG5cbmltcG9ydCB4IGZyb20gJ21vZHVsZSc7XG5cbmltcG9ydCAqIGFzIHkgZnJvbSAnbW9kdWxlJzsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)). Both are just local names, so if the rule checks one then I don't think there's any reason it shouldn't check the other.

Since this change would produce more warnings, I'd like more opinions from @eslint/eslint-tsc before marking it as accepted.

**Comment by nzakas** [comment]
> This rule already checks default imports ([Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgaWQtbGVuZ3RoOiBbXCJlcnJvclwiXSovXG5cbmltcG9ydCAqIGFzIHggZnJvbSAnbW9kdWxlJztcblxuaW1wb3J0IHkgZnJvbSAnbW9kdWxlJzsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)). Both are just local names, so if the rule checks one then I don't think there's any reason it shouldn't check the other.

Agree with this logic. Seems like a bug. Marking as accepted.
